src/views/Dashboard/components/Widgets/SendWidget/SendWidget.tsx

```diff
@@ -0,0 +1,191 @@
+    if (getAccount !== undefined && getAccountId !== undefined) {
+      try {
+        const result = await getAccount(toAddress);
+        if (result) {
+          const accountResult = await getAccountId(result.publicKey);
+          if (accountResult) {
+            return accountResult.account;
+          }
+        }
+      } catch (e) {
+        console.error("error while fetching public key:", e);
+        return;
+      }
+    }
+  }, [getAccount, getAccountId, toAddress]);
+
+  // need a final useEffect which gets run when all other pieces are ready to build the transaction
+  const prepareUnsignedTx = useCallback(
+    async (secret: string): Promise<IUnsignedTransaction | undefined> => {
+      // make sure address is valid
+      if (!isValidAddress(toAddress)) {
+        return;
+      }
+
+      if (accountRs === undefined || sendQuantity === undefined) {
+        return;
+      }
+
+      const recipientAccountId = await fetchRecipAccountId();
+      const tx: IUnsignedTransaction = {
+        senderPublicKey: publicKey, // publicKey from useAccount() hook
+        senderRS: accountRs, // accountRs from useAccount() hook
+        // sender: "123", // required in some situations?
+        feeNQT: standardFee,
+        version: 1,
+        phased: false,
+        type: 0,
+        subtype: 0,
+        attachment: { "version.OrdinaryPayment": 0 },
+        amountNQT: sendQuantity, // TODO: write converter function
+        recipientRS: toAddress,
+        recipient: recipientAccountId,
+        ecBlockHeight: 0, // must be included
+        deadline: standardDeadline,
+        timestamp: Math.round(Date.now() / 1000) - JUPGenesisTimestamp, // Seconds since Genesis. sets the origination time of the tx (since broadcast can happen later).
+        secret: secret,
+      };
+
+      return tx;
+    },
+    [accountRs, fetchRecipAccountId, publicKey, sendQuantity, toAddress]
+  );
+
+  const handleSend = useCallback(async () => {
+    if (sendJUP !== undefined) {
+      setRequestUserSecret(true);
+    }
+  }, [sendJUP]);
+
+  const handleSubmitSecret = useCallback(
+    async (secret: string) => {
+      const unsignedTx = await prepareUnsignedTx(secret);
+      if (sendJUP !== undefined && unsignedTx !== undefined) {
+        const result = await sendJUP(unsignedTx);
+        console.log("send result:", result);
+      }
+    },
+    [prepareUnsignedTx, sendJUP]
+  );
+
+  const handleToAddressEntry = useCallback(
+    (toAddressInput: string) => {
+      setToAddress(toAddressInput);
+    },
+    [setToAddress]
+  );
+
+  const handleQuantityEntry = useCallback((quantityInput: string) => {
+    setSendQuantity(quantityInput);
+  }, []);
+
+  const handleSecretEntry = useCallback((secretInput) => {
+    setUserSecretInput(secretInput);
+  }, []);
+
+  // MUST: currently the widget disappears when the dialog appears, it would be nice if it stayed in the background
+  return requestUserSecret ? (
```

Comment:
I'd suggest that for readability you pull the following logic out into a useMemo. it'll also give a small increase in performance but that's quite minimal.

you can choose to ignore this, it's more of a code readability suggestion than performance.